Subject: [PATCH] Fix building against the system portaudio library

Building against the system portaudio results in this error:

./src/AudioIO.cpp:983: undefined reference to `PaUtil_GetTime'
audacity-AudioIO.o: In function `audacityAudioCallback(void const*, void*,
unsigned long, PaStreamCallbackTimeInfo const*, unsigned long, void*)':
./src/AudioIO.cpp:4630: undefined reference to `PaUtil_GetTime'
collect2: error: ld returned 1 exit status
Makefile:2349: recipe for target 'audacity' failed
make[3]: *** [audacity] Error 1

This is because PaUtil_GetTime is declared as a C symbol in pa_util.h
but is resolved as a C++ symbol at link time.

Audacity fixes this in the local tree with this change:
https://github.com/audacity/audacity/commit/38fd97b8e26060332ab3e9e000a8882326a70ba7

However this is not general enough for the portaudio debian package.

Since PaUtil_GetTime() is the only function causing problems, just copy
over the code where it's used.
---
 src/AudioIO.cpp | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

--- audacity-minsrc-2.3.3/src/AudioIO.cpp.orig	2019-11-25 18:57:25.820000000 +0100
+++ audacity-minsrc-2.3.3/src/AudioIO.cpp	2019-11-25 18:59:07.160000000 +0100
@@ -480,8 +480,23 @@
    #define ROUND(x) (int) ((x)+0.5)
    //#include <string.h>
 //   #include "../lib-src/portmidi/pm_common/portmidi.h"
-   #include "../lib-src/portaudio-v19/src/common/pa_util.h"
    #include "NoteTrack.h"
+   #include <sys/time.h>
+	PaTime PaUtil_GetTime( void )
+	{
+		#ifdef HAVE_MACH_ABSOLUTE_TIME
+			    return mach_absolute_time() * machSecondsConversionScaler_;
+		#elif defined(HAVE_CLOCK_GETTIME)
+		    struct timespec tp;
+		    clock_gettime(CLOCK_REALTIME, &tp);
+		    return (PaTime)(tp.tv_sec + tp.tv_nsec * 1e-9);
+		#else
+		    struct timeval tv;
+		    gettimeofday( &tv, NULL );
+		    return (PaTime) tv.tv_usec * 1e-6 + tv.tv_sec;
+		#endif
+	}
+
 #endif
 
 #ifdef EXPERIMENTAL_AUTOMATED_INPUT_LEVEL_ADJUSTMENT
